README.md

```diff
@@ -143,15 +143,28 @@ still apply)
 
 ### API
 
-- `compileJS`: async `{ component, error, name }` - compiles a single JS file
+#### Methods
+
+- `compileJS`: async, returns `compileResult` - compiles a single JS file
    uses the syntax from [ember-template-imports](https://github.com/ember-template-imports/ember-template-imports)
-- `compileHBS`: `{ component, error, name }` - compiles a template-only component with no dependencies
+- `compileHBS`: returns `compileResult` - compiles a template-only component with no dependencies
 - `invocationOf`: `string` - converts hyphenated text to an `<AngleBracketInvocation />`
 - `nameFor`: `string` - generates a component-safe GUID-like derivation from code
 
-_`component`_: invokable from templates
-_`error`_: if there is a compilation error, this will be non-falsey
-_`name`_: the name assigned to the input text via UUIDv5
+#### Properties
+
+```ts
+interface CompileResult {
```

Comment:
The description comments might be better in a separate bullet points after the interface definition once there are more words. Right now it seems reasonable to inline them.